src/handlers/profile.ts

```diff
@@ -21,6 +30,25 @@ export default function profileHandler(sessionStore: ISessionStore) {
       return;
     }
 
+    if (options && options.refetch) {
+      if (!session.accessToken) {
+        throw new Error('The access token needs to be saved in the session for the user to be fetched');
+      }
+
+      const { body: user } = await getAsync({
+        baseUrl: `https://${settings.domain}`,
+        url: 'userinfo',
+        json: true,
+        headers: {
+          Authorization: `Bearer ${session.accessToken}`
+        }
+      });
+
+      await sessionStore.save(req, res, { ...session, user });
```

Comment:
Can we merge the current user in the session with the new claims we get from userinfo instead of replacing the whole user? There are cases where some claims are in the id_token but when they are not returned as part of userinfo. If that happens, we don't want to break the current session. With userinfo we just want to make sure we can get the newest info like updated email address or email_verified etc... and update that in the session.

Reply:
I am not sure how to merge claims.  Could you help with that?

Reply:
@sandrinodimattia Could you help with the merging of claims?  I think that is the only thing holding this back.  It would be great to get this merged in so everyone could benefit from it.

Reply:
Ok looking into it!